packages/cli/src/metadataGeneration/metadataGenerator.ts

```diff
@@ -111,6 +112,67 @@ export class MetadataGenerator {
     }
   };
 
+  private checkForPathParamSignatureDuplicates = (controllers: Tsoa.Controller[]) => {
+    const controllerDup: { [key: string]: { [key: string]: Tsoa.Method[] } } = {};
+    let message = '';
+
+    controllers.forEach(controller => {
+      const chunks: { [key: string]: Tsoa.Method[] } = {};
+      controller.methods.forEach(method => {
+        if (chunks[method.method] === undefined) {
+          chunks[method.method] = [];
+        }
+        chunks[method.method].push(method);
+      });
+
+      const chunkDupPath: { [key: string]: Tsoa.Method[] } = {};
+      const regex = new RegExp('^{.*?}/?');
```

Comment:
tsoa also accept `/:id` instead of `/{id}` for templating. Can you adapt the Regex to match this templating syntax aswell?

Reply:
@WoH 
Thank you! I did forget to handle `/:id`.  

I also add a warn log when some partial route is duplicate like:
```
a/b/{c}
a/b
```
more test cases also added.